packages/rum-core/src/domain/rumEventsCollection/view/trackInteractionToNextPaint.ts

```diff
@@ -0,0 +1,142 @@
+import { find, type Duration, noop } from '@datadog/browser-core'
+import type { BrowserWindow, RumEventTiming, RumFirstInputTiming } from '../../../browser/performanceCollection'
+import { LifeCycleEventType, type LifeCycle } from '../../lifeCycle'
+import { ViewLoadingType } from '../../../rawRumEvent.types'
+
+// Arbitrary value to prevent unnecessary memory usage on views with lots of interactions.
+const MAX_INTERACTION_ENTRIES = 10
+
+// List of longest interactions on the view by duration.
+// The list is as most MAX_INTERACTION_ENTRIES long.
+const longestInteractions: Array<RumEventTiming | RumFirstInputTiming> = []
+
+/**
+ * Track the interaction to next paint (INP).
+ * To avoid outliers, return the p98 worst interaction of the view.
+ * Documentation: https://web.dev/inp/
+ * Reference implementation: https://github.com/GoogleChrome/web-vitals/blob/main/src/onINP.ts
+ */
+export function trackInteractionToNextPaint(
+  viewLoadingType: ViewLoadingType,
+  lifeCycle: LifeCycle,
+  callback: (layoutShift: Duration) => void
+) {
+  const { getInteractionCount, stop: stopInteractionCountTracking } = trackInteractionCount(viewLoadingType)
+  let maxInpDuration = 0 as Duration
+
+  const { unsubscribe: stop } = lifeCycle.subscribe(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, (entries) => {
+    for (const entry of entries) {
+      if (
+        (entry.entryType === 'event' && entry.interactionId) ||
+        // Entries of type `first-input` don't currently have an `interactionId`,
+        // so to consider them in INP we have to first check that an existing
```

Comment:
Hum, good question. FMU, first-input is reported as  performance event entry if it is above durationThreshold. Since the durationThreshold is at most 16ms, if the first input is below that value it won't be reported as a performance event entry.
It's true that in our case the durationThreshold is 40 so this scenario will never happen. I guess Google does this check because in their repo the [durationThreshold is an option](https://github.com/GoogleChrome/web-vitals/blob/3806160ffbc93c3c4abf210a167b81228172b31c/src/onINP.ts#L209C32-L209C49) the developer can change which could go bellow 16.

Reply:
I've removed this logic since it's not necessary for us

Reply:
Mmh `first-input` entries are not `event` entries (that's two different types). The reference implementation [observes `first-input` separately](https://github.com/GoogleChrome/web-vitals/blob/main/src/onINP.ts#L224), so I guess it is collected regardless of the duration threshold?